### PR TITLE
Shell implementation for RandomUniform.

### PR DESCRIPTION
**Pull request by popovaan**
### Details:
 - *Shell implementation for RandomUniform.*

### Tickets:
 - *56595*

### CODE REVIEW

File: ngraph/core/include/ngraph/op/random_uniform.hpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                /// \param      max_val           Node producing the tensor with maximum value.
+                /// \param      out_type          Output type of the tensor.
+                /// \param      global_seed       Global seed value.
+                /// \param      op_seed           Operational seed value.
+                RandomUniform(const Output<Node>& out_shape,
+                              const Output<Node>& min_val,
+                              const Output<Node>& max_val,
+                              const ngraph::element::Type& out_type,
+                              int64_t global_seed,
+                              int64_t op_seed);
```
Comment (lazarevevgeny) [inline ngraph/core/include/ngraph/op/random_uniform.hpp:38]: `uint64_t`?

File: ngraph/core/src/op/random_uniform.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v8::RandomUniform, "RandomUniform", 8);
+
+op::v8::RandomUniform::RandomUniform(const Output<Node>& out_shape,
+                                     const Output<Node>& min_val,
+                                     const Output<Node>& max_val,
+                                     const ngraph::element::Type& out_type,
+                                     int64_t global_seed,
+                                     int64_t op_seed)
```
Comment (lazarevevgeny) [inline ngraph/core/src/op/random_uniform.cpp:19]: `uint64_t`?

File: ngraph/core/include/ngraph/op/random_uniform.hpp (hunk 1 above)
Comment (popovaan) [reply ngraph/core/include/ngraph/op/random_uniform.hpp:38]: uint32_t can be used

Comment (popovaan) [reply ngraph/core/include/ngraph/op/random_uniform.hpp:38]: Yes, mistake.

**Review by lazarevevgeny — APPROVED** [review]

**Review by vgavrilo — APPROVED** [review]

**Review by yekruglov — APPROVED** [review]

**Comment by rkazants** [comment]
@ilyachur, please take a look at RandomUniform shell implementation

**Review by rkazants — APPROVED** [review]

File: ngraph/core/src/op/random_uniform.cpp [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                                      min_val < max_val,
+                                      "Min value must be less than max value. Got "
+                                      "min value: ",
+                                      min_val,
+                                      ", max value: ",
+                                      max_val);
+            }
+            else
+            {
+                double min_val = const_min->cast_vector<double>()[0];
```
[the remark this answers is not in the document]
Comment (popovaan) [reply ngraph/core/src/op/random_uniform.cpp:107]: According to specification of random uniform min and max value can be int32, int64, double, float32 and float16. So I didn't consider int8.

File: ngraph/core/src/op/random_uniform.cpp [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+void op::v8::RandomUniform::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v8_RandomUniform_validate_and_infer_types);
+
+    const auto& shape_et = get_input_element_type(0);
+    NODE_VALIDATION_CHECK(this,
+                          shape_et.is_dynamic() || shape_et == element::i32 ||
+                              shape_et == element::i64,
+                          "Output shape must have int32 or int64 element type.");
```
[the remark this answers is not in the document]
Comment (popovaan) [reply ngraph/core/src/op/random_uniform.cpp:36]: 0 input of random uniform is tensor which defines output shape. So I check element type of this tensor with output shape. Is there any mistake?

File: ngraph/core/src/op/random_uniform.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (popovaan) [reply ngraph/core/src/op/random_uniform.cpp:107]: It looks like I need to add type check here and raise error if type is not supported.

[the remark this answers is not in the document]
Comment (popovaan) [reply ngraph/core/src/op/random_uniform.cpp:110]: Done.

File: ngraph/core/src/op/random_uniform.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (popovaan) [reply ngraph/core/src/op/random_uniform.cpp]: Corrected error message.

File: ngraph/core/include/ngraph/op/random_uniform.hpp [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                /// \return The output tensor type.
+                const ngraph::element::Type& get_out_type() const { return m_output_type; }
+                void set_out_type(const ngraph::element::Type& output_type)
+                {
+                    m_output_type = output_type;
+                }
+
+                /// \return The global seed value.
+                uint64_t get_global_seed() const { return m_global_seed; }
+                void set_global_seed(const uint64_t& seed) { m_global_seed = seed; }
```
[the remark this answers is not in the document]
Comment (popovaan) [reply ngraph/core/include/ngraph/op/random_uniform.hpp]: It is not needed. Removed reference.

File: ngraph/core/src/op/random_uniform.cpp [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                                      "Min value must be less than max value. Got "
+                                      "min value: ",
+                                      min_val,
+                                      ", max value: ",
+                                      max_val);
+            }
+            else if (get_out_type() == ngraph::element::Type_t::f16 ||
+                     get_out_type() == ngraph::element::Type_t::f32 ||
+                     get_out_type() == ngraph::element::Type_t::f64 ||
+                     get_out_type() == ngraph::element::Type_t::bf16)
```
Comment (ilyachur) [inline ngraph/core/src/op/random_uniform.cpp:108]: If I remember right, we can use `is_real()` here. I propose to do it separately.

**Review by ilyachur — APPROVED** [review]

File: ngraph/core/src/op/random_uniform.cpp (hunk 6 above)
Comment (popovaan) [reply ngraph/core/src/op/random_uniform.cpp]: Done.

**Pull request merged**